README.md

```diff
@@ -1,19 +1,75 @@
-# learning-haskell
+# Learning Haskell at [Shuttl](http://shuttl.com/)
 
-Accompnaying homework assignments for the lecture series done at Shuttl by me.
+###### A Lecture Series by [Sherub Thakur](https://github.com/sherubthakur), a Haskell Lover! 
 
-# Setup
-You will need to have haskell setup for you to work on it. There are many ways to go about doing it I will share a beginner friendly way to set this up
 
-## Setup Haskell on your machine
-1. Get Stack `curl -sSL https://get.haskellstack.org/ | sh`
-2. `stack ghci`
-3. Try out `1 + 1` in the repl
-Note: https://docs.haskellstack.org/en/stable/README/#how-to-install
+## Prerequisite Setup
+To work with _Haskell_, we'll first need to set it up. Further, there're multiple ways to go about doing it. 
+Let's start with a simple one, which is friendly for beginner programmers:
+
+### stack Installation
+We can use a tool called _stack_ to work on our _Haskell_ projects. So, of course, we need to install it before we can go any further.
+
+One easy way to install _stack_ involves executing a _.sh_ script:
+```shell script
+$ curl -sSL https://get.haskellstack.org/ | sh
+```
+
+Well, there's a good reason for using the _-L_ flag with our [cURL](http://www.mit.edu/afs.new/sipb/user/ssen/src/curl-7.11.1/docs/curl.html) here. That's because the URL used here responds with a 302 (redirection) HTTP status code, and we need to follow through the redirection.
+
+To dive deeper into this concept, we can look at the header information by using the -D flag:
+
+```shell script
+$ curl -is -o /dev/null -D - https://get.haskellstack.org/
+
+HTTP/2 302
+server: nginx/1.17.8
+date: Mon, 20 Apr 2020 15:55:07 GMT
+content-type: text/html
+content-length: 145
+location: https://raw.githubusercontent.com/commercialhaskell/stack/stable/etc/scripts/get-stack.sh
+```
+
+With the header information available to us, We can now see the actual location of our script.
```

Comment:
Can we shift this part into FAQ. I would like the readme to be like a no-brainer thing first and foremost. And if folks are interested in knowing more let's address this using an FAQ section.

Reply:
@sherubthakur  Done 🙂